### PR TITLE
DEV-3237 Upgrade orange to 2.4

### PR DESCRIPTION
**Pull request by pauldwolfe**
Version bump and add the disclaimer when RUO.

### CODE REVIEW

File: cluster/src/test/java/com/hartwig/pipeline/tertiary/orange/OrangeTest.java [hunk 1]
```diff
@@ -66,38 +73,63 @@ protected List<String> expectedInputs() {
...
+                TestInputs.linxGermlineOutput(),
+                TestInputs.linxSomaticOutput(),
+                TestInputs.cuppaOutput(),
+                TestInputs.virusInterpreterOutput(),
+                TestInputs.peachOutput(),
+                TestInputs.sigsOutput(),
+                TestInputs.REF_GENOME_37_RESOURCE_FILES,
+                Pipeline.Context.RESEARCH);
+        List<BashCommand> commands = victim.tumorReferenceCommands(TestInputs.defaultSomaticRunMetadata());
+        assertThat(commands.get(2).asBash()).contains("-add_disclaimer");
```
Comment (nedleitch) [inline cluster/src/test/java/com/hartwig/pipeline/tertiary/orange/OrangeTest.java:132]: I don't love the use of the magic number, what about making it clear it's the last element using `commands.get(commands.size() - 1)`?

**Review by nedleitch — CHANGES_REQUESTED** [review]

File: cluster/src/main/java/com/hartwig/pipeline/SomaticPipeline.java [hunk 2]
```diff
@@ -219,7 +220,7 @@ public PipelineState run(final AlignmentPair pair) {
                                         linxGermlineOutput,
                                         linxSomaticOutput,
                                         cuppaOutput,
-                                        virusInterpreterOutput, peachOutput, sigsOutput, resourceFiles)));
+                                        virusInterpreterOutput, peachOutput, sigsOutput, resourceFiles, arguments.context())));
```
Comment (kduyvesteyn) [inline cluster/src/main/java/com/hartwig/pipeline/SomaticPipeline.java:223]: Formatting looks pretty arbitrary!

File: cluster/src/main/java/com/hartwig/pipeline/tools/Versions.java [hunk 3]
```diff
@@ -21,7 +21,7 @@ public interface Versions {
     String HEALTH_CHECKER = "3.4";
     String LILAC = "1.4.2";
     String LINX = "1.23.2";
-    String ORANGE = "2.3";
+    String ORANGE = "2.4";
```
Comment (kduyvesteyn) [inline cluster/src/main/java/com/hartwig/pipeline/tools/Versions.java:24]: You should use 2.4.0

Related: I don't understand where the "2.4" jar comes from on data-vm, there is no "2.4" release.

**Review by kduyvesteyn — COMMENTED** [review]
Apart from the comments in this review, there are some further requirements for being able to use ORANGE produced by pipeline in ACTIN (https://hartwigmedical.atlassian.net/browse/ACTIN-99)

File: cluster/src/test/java/com/hartwig/pipeline/tertiary/orange/OrangeTest.java (hunk 1 above)
Comment (pauldwolfe) [reply cluster/src/test/java/com/hartwig/pipeline/tertiary/orange/OrangeTest.java:132]: One way or another it's a magic number. I am actually looking for the third and not the second to last. I could also pluck out the orange command by filtering on the orange jar or something but then am i really adding value? As a compromise I'll assign to a descriptive variable such that its clear for others what we're doing here

**Pull request merged**